compiler-core/templates/prelude.js

```diff
@@ -316,6 +316,10 @@ export function isEqual(x, y) {
       unequalSets(a, b);
     if (unequal) return false;
 
+    if (typeof a.equals === "function" && a.equals(b)) {
+      continue;
```

Comment:
I think we want to return the value of `a.equals(b)` here

Reply:
Ah yea, I think its if false return, if true continue? Since there could be more values to check on the stack. Does that sound right?

Reply:
Yeah if we return true early then in
```
{ x: { equals () { … } }
, y: “foo”
}
```
we’d never check `y` for equality, seems like the wrong behaviour to me. 

We don’t return true early for identity equality I don’t see why this would be any different!